### PR TITLE
Fixes for GHCJS frontend builds

### LINKED ISSUE #6: Frontend build failing to enter nix-shell
Could not enter reflex/ghcjs nix-shell for TAS frontend build using data-validation. Final error message not informative to me, but suspect regex-tdfa https://hackage.haskell.org/package/regex-tdfa. 
```shell
Build with /nix/store/47fs1p9v07wnqp044ldp22i79njw6r2j-ghcjs-8.6.0.1.
unpacking sources
unpacking source archive /nix/store/wnhbyj3x1h7z16grwji42wzcdlb9lmp0-source/.
unpacker produced multiple directories
builder for '/nix/store/39ir4755pb52ri1lfr21praw7fx7shqm-data-validation-0.1.0.0.drv' failed with exit code 1
cannot build derivation '/nix/store/83nbqfqgvfl8nrgld7jrmcs8jfrc4r7l-ghcjs-8.6.0.1-with-packages.drv': 1 dependencies couldn't be built
error: build of '/nix/store/83nbqfqgvfl8nrgld7jrmcs8jfrc4r7l-ghcjs-8.6.0.1-with-packages.drv' failed   
```

### PR DESCRIPTION
**Pull request by sambolias**
See #6 for details on failing ghcjs builds. This PR suggests edits that 
- remove baked in regex-tdfa lib usage, moves into example to show how any regex lib can be leveraged using `disputeWithFact` 
  - this seems like a good idea to not depend on regex libs that do not work well universally
- moves Internal.hs from internal library to main
  - this doesn't seem best option because it does expose the internal code, but as issue shows it would not build. this must be a bug with ghcjs or reflex-platform and is the reason why this PR is a draft

This also fixes a couple type name mismatches in example, and removes a lot of empty whitespace (makes it harder to read diff but my text editor does this auto so hard to avoid)

closes #6

### CODE REVIEW

File: data-validation.cabal [hunk 1]
```diff
@@ -10,34 +10,28 @@ extra-source-files:  CHANGELOG.md, README.md
 
 library
   exposed-modules:    Data.Validation
+                    , Data.Validation.Internal
```
Comment (devinlyons) [inline data-validation.cabal:13]: Why did you eliminate the internal module? It doesn't have anything to do with this issue.

**Review by devinlyons — COMMENTED** [review]
I agree with the change in general. Removing the regex dependency is fine with me. However, there were other changes I don't understand.

File: data-validation.cabal (hunk 1 above)
Comment (boj) [reply data-validation.cabal:13]: The whole point of this PR is because of issue #6

Comment (devinlyons) [reply data-validation.cabal:13]: Ah, I didn't see the update to the issue. Well, if ghcjs won't support internal modules, I guess there's nothing else we can do.

**Review by devinlyons — APPROVED** [review]

File: data-validation.cabal (hunk 1 above)
Comment (wduncanfraser) [reply data-validation.cabal:13]: Feels like something funky is going on here, as we definitely consume libs with multiple modules. And our exe vs lib setup is effectively this as well.

Comment (boj) [reply data-validation.cabal:13]: Sam is still trying to discover what is going on here, which is why this is a Draft PR. If he can't move past this, then it's a backup plan to unblock him.

**Comment by sambolias** [comment]
@devinlyons I am going to work with this a bit more in case there are any other frontend edge cases. I'll also do a little more research on the internal library ghcjs problem, because that should work. But if I cannot find any better way I will merge this as it does fix the issues.

**Review by boj — APPROVED** [review]
Sam talked to Ryan Trinkle about this, and it sounds like the answer was "we had to flatten things." Ryan is the final answer on Reflex-anything, so it looks like we need to run with this PR.

**Comment by sambolias** [comment]
Yes from IRC conversation I think flattening it is a viable solution. Here is a snippet of conversation in case we ever decide to try troubleshooting the sublibs, should be good place to start. 

```
sambolias 15:47:42
I'm troubleshooting an issue with ghcjs reflex-platform frontend build failing with a dependency that uses cabal sub-libraries like https://www.haskell.org/cabal/users-guide/developing-packages.html#sublibs. I can flatten libraries in dependency and frontend build works, but sublibs configuration only works for ghc builds. My specific error can be seen https://github.com/alasconnect/data-validation/issues/6#issuecomment-632265084. Is there any known issues between cabal sublibs and ghcjs or reflex-platform?

ryantrinkle 15:52:16
sambolias: i think i've run into a similar issue.  my solution was to flatten it

ryantrinkle 15:52:26
it seems like ghci and ghc-pkg disagree about which package is which

ryantrinkle 15:52:36
maybe other things also disagree
in my case, it looked as though ghci thought the main library was hidden, when really it was only the sub-library that was hidden
whereas ghc-pkg correctly knew which one was hidden
```

**Pull request merged**